dashboard/app/models/user.rb

```diff
@@ -836,7 +836,7 @@ def User.track_proficiency(user_id, script_id, level_id)
 
   # returns whether a new level has been completed and asynchronously enqueues an operation
   # to update the level progress.
-  def track_level_progress_async(script_level, new_result, submitted)
+  def track_level_progress_async(script_level, new_result, submitted, level_source_id)
```

Comment:
Can you make level_source_id default to nil so you don't have to change user_script.rb?